src/inmanta/execute/runtime.py

```diff
@@ -310,24 +347,31 @@ class DelayedResultVariable(ResultVariable[T]):
 
     def __init__(self, queue: "QueueScheduler", value: Optional[T] = None) -> None:
         ResultVariable.__init__(self, value)
-        self.promises: List[IPromise] = []
-        self.done_promises: Set[IPromise] = set()
+        self.promises: Optional[List[IPromise]] = []
+        self.done_promises: Optional[Set[IPromise]] = set()
         self.queued = False
         self.queues = queue
         if self.can_get():
             self.queue()
 
     def get_promise(self, provider: "Statement") -> ISetPromise[T]:
         promise: ISetPromise[T] = SetPromise(self, provider)
-        self.promises.append(promise)
+        if self.promises is not None:
+            # only track the promise if this variable has not been frozen yet.
+            self.promises.append(promise)
         return promise
 
     def get_progression_promise(self, provider: "Statement") -> ProgressionPromise:
+        if self.promises is None:
+            return None
```

Comment:
I guess this method is not typed correctly?

Reply:
Indeed, the parent has the correct type but I forgot to change this one.